From 27fe5e0a4429c0a92635d1395a673fcea6289046 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Povi=C5=A1er?= <povik+lin@cutebit.org>
Date: Sat, 19 Feb 2022 09:49:50 +0100
Subject: [PATCH 076/171] ASoC: cs42l42: Bypass device ID check

The cs42l42 driver is also applicable to the cs42l83 part.
---
 sound/soc/codecs/cs42l42.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/sound/soc/codecs/cs42l42.c b/sound/soc/codecs/cs42l42.c
index 5fb28df85b8a..2a98f9bc7144 100644
--- a/sound/soc/codecs/cs42l42.c
+++ b/sound/soc/codecs/cs42l42.c
@@ -2273,13 +2273,10 @@ static int cs42l42_i2c_probe(struct i2c_client *i2c_client)
 		goto err_disable;
 	}
 
-	if (devid != CS42L42_CHIP_ID) {
-		ret = -ENODEV;
-		dev_err(&i2c_client->dev,
+	if (devid != CS42L42_CHIP_ID)
+		dev_warn(&i2c_client->dev,
 			"CS42L42 Device ID (%X). Expected %X\n",
 			devid, CS42L42_CHIP_ID);
-		goto err_disable;
-	}
 
 	ret = regmap_read(cs42l42->regmap, CS42L42_REVID, &reg);
 	if (ret < 0) {
-- 
2.34.1

